Automattic/wp-calypso
# Hooks: Add pre-commit hook for PHPCS.

Now that we have PHP files in our `/apps` directory, let's have PHPCS run automatically as a pre-commit hook.

It requires that PHPCS is available locally, and will not attempt to process PHP files without it (but will continue with linting everything else). The custom ruleset defined at `apps/phpcs.xml` is used for both PHPCBF and PHPCS.

#### Testing instructions

* Apply this branch locally.
* Modify a PHP file within `/apps/full-site-editing` that includes linting problems (eg. too many spaces, unused variables).
* From the command-line at your `wp-calypso` root, run `node bin/pre-commit-hook.js`.
* Verify that the affected file was cleaned up, and issues were reported.
* Do funny things; try to break it, or cause unexpected behaviour.
